hardware/nao/nao_service.py

```
@@ -0,0 +1,229 @@
+        # Test if NAO is reachable first (quick timeout)
+        test_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        test_socket.settimeout(2.0)
+        result = test_socket.connect_ex((NAO_IP, PORT))
+        test_socket.close()
+        
+        if result != 0:
+            print("NAO robot not reachable at", NAO_IP, ":", PORT)
+            return False
+        
+        # If reachable, try to create proxy
+        test_proxy = ALProxy("ALBehaviorManager", NAO_IP, PORT)
```

Ruff (0.14.2)

Local variable `test_proxy` is assigned to but never used

Remove assignment to unused variable `test_proxy`

(F841)